pkg/scheduler/factory_test.go

```diff
@@ -137,6 +138,21 @@ func TestCreateFromConfig(t *testing.T) {
 	if string(encoding) != want {
 		t.Errorf("Config for NodeLabel plugin mismatch. got: %v, want: %v", string(encoding), want)
 	}
+
+	// Verify that service affinity predicates are converted to framework plugins.
+	if _, ok := findPlugin(serviceaffinity.Name, "FilterPlugin", conf); !ok {
+		t.Fatalf("ServiceAffinity plugin not exist in framework.")
+	}
+	// Verify that the policy config is converted to plugin config for service affinity predicate.
+	serviceAffinityConfig := findPluginConfig(serviceaffinity.Name, conf)
+	encoding, err = json.Marshal(serviceAffinityConfig)
+	if err != nil {
+		t.Errorf("Failed to marshal %+v: %v", serviceAffinityConfig, err)
+	}
+	want = `{"Name":"ServiceAffinity","Args":{"labels":["zone"]}}`
```

Comment:
Please update the test case to include multiple policy config for the ServiceAffinity and check that the labels are aggregated.

Reply:
This is not addressed. We should update the tests.

Reply:
Sorry, was looking at old version for some reason. 
/LGTM.